File: dashboard/src/types/strategy.ts
export type ParamType = 'number' | 'string' | 'boolean' | 'select'

export type ParamSchema = {
  key: string
  label: string
  type: ParamType
  default: unknown
  min?: number
  max?: number
  options?: string[]
}

export type StrategySchema = {
  slug: string
  display_name: string
  params: ParamSchema[]
}

export type StrategyNode = {
  strategy_slug: string
  weight: number
  params: Record<string, unknown>
}

export type CreateStrategy = {
  name: string
  display_name: string
  threshold: number
  nodes: StrategyNode[]
}

export type CompositeConfig = {
  threshold: number
  nodes: StrategyNode[]
}

export type Strategy = {
  slug: string
  display_name: string
  is_composite: boolean
  created_at: string
  config?: CompositeConfig
}

File: trader-core/internal/db/models/strategy.go
package models

import (
	"time"

	"gorm.io/datatypes"
)

type StrategyNode struct {
	StrategySlug string  `json:"strategy_slug"`
	Weight       float64 `json:"weight"`
}

type Strategy struct {
	Slug          string         `gorm:"primaryKey"`
	DisplayName   string         `gorm:"not null"`
	IsComposite   bool           `gorm:"not null;default:false"`
	DefaultConfig datatypes.JSON `gorm:"type:jsonb;not null;default:'{}'"`
	Nodes         datatypes.JSON `gorm:"type:jsonb;not null;default:'[]'"`
	CreatedAt     time.Time
	UpdatedAt     time.Time
}
